scipy/stats/_continuous_distns.py

```diff
@@ -1451,7 +1451,24 @@ def _shape_info(self):
 
     def _pdf(self, x):
         # cauchy.pdf(x) = 1 / (pi * (1 + x**2))
-        return 1.0/np.pi/(1.0+x*x)
+        with np.errstate(over='ignore'):
+            return 1.0/np.pi/(1.0+x*x)
+
+    def _logpdf(self, x):
+        # The formulas
+        #     log(1/(pi*(1 + x**2))) = -log(pi) - log(1 + x**2)
+        #                            = -log(pi) - log(x**2*(1 + 1/x**2))
+        #                            = -log(pi) - (2log(|x|) + log1p(1/x**2))
+        # are used here.
+        absx = np.abs(x)
```

Comment:
I checked, and it's probably best to keep this as you have it - always calculate `absx` so it can be used in the `cond` and in `f2`. The alternative - `cond = (x < 1) & (x > -1)` and calculating `np.abs(x)` only where needed - is not clearly better.